recipe/meta.yaml

```diff
@@ -65,56 +80,99 @@ build:
+    - cusparselt  # [(linux64 or win) and cuda_compiler_version == "11.2"]
 
 requirements:
   build:
     - {{ compiler("c") }}
     - {{ compiler("cxx") }}
     - {{ compiler("cuda") }}
-    - sysroot_linux-64 2.17  # [linux64]
+    #- cuda-nvcc
+    - sysroot_{{ target_platform }} 2.17    # [linux]
+    - cross-python_{{ target_platform }}    # [build_platform != target_platform]
+    - python                                # [build_platform != target_platform]
+    - cython >=0.29.22,<3                   # [build_platform != target_platform]
+    # TODO: clean up
+    {% if cuda_major_minor == (12, 0) %}
+    - cuda-driver-dev                       # [build_platform != target_platform]
+    - cuda-cudart-dev                       # [build_platform != target_platform]
+    - cuda-nvrtc-dev                        # [build_platform != target_platform]
+    - cuda-nvtx-dev                         # [build_platform != target_platform]
+    - cuda-profiler-api                     # [build_platform != target_platform]
+    - cuda-cccl_{{ build_platform }}        # [build_platform != target_platform]
+    - libcublas-dev                         # [build_platform != target_platform]
+    - libcufft-dev                          # [build_platform != target_platform]
+    - libcurand-dev                         # [build_platform != target_platform]
+    - libcusolver-dev                       # [build_platform != target_platform]
+    - libcusparse-dev                       # [build_platform != target_platform]
+    {% endif %}
+    # optional dependencies for CUDA 11.2+
+    - cudnn >=8                             # [build_platform != target_platform and (not linux64 and cuda_compiler_version != "12.0")]
+    - nccl >=2.8                            # [build_platform != target_platform and cuda_compiler_version != "12.0"]
+    - cutensor >=1.4                        # [build_platform != target_platform]
 
   host:
     - python
     - pip
     - setuptools
     - cython >=0.29.22,<3
     - fastrlock >=0.5
-    - cudnn >=7.6
-    - nccl >=2.8  # [linux]
-    - cutensor >=1.4
-    {% if cuda_major_minor >= (11, 2) %}
-    - cusparselt 0.2.0.*  # [linux64 or win]
+    - cuda-version {{ cuda_compiler_version }}
+    # TODO: clean up
+    {% if cuda_major_minor == (12, 0) %}
+    - cuda-driver-dev
+    - cuda-cudart-dev
+    - cuda-nvrtc-dev
+    - cuda-nvtx-dev
+    - cuda-profiler-api
+    - cuda-cccl_{{ target_platform }}
+    - libcublas-dev
+    - libcufft-dev
+    - libcurand-dev
+    - libcusolver-dev
+    - libcusparse-dev
     {% endif %}
+    # optional dependencies
+    # TODO: see https://github.com/conda-forge/cudnn-feedstock/issues/58
+    - cudnn >=8           # [not ((aarch64 or ppc64le) and cuda_compiler_version == "12.0")]
+    # TODO: wait for NCCL to catch up
+    - nccl >=2.8          # [linux and cuda_compiler_version != "12.0"]
+    - cutensor >=1.4,<2
+    - cusparselt 0.2.0.*  # [(linux64 or win) and cuda_compiler_version == "11.2"]
```

Comment:
Wonder if we can drop some of these selectors now that these packages have been rebuilt